src/wield/model/optics/mirror.py

```diff
@@ -324,6 +335,15 @@ def visit_mode_matching_transport(self, manip):
         depth_m = manip.p["depth[m]"]
         ROC_B_m = manip.p["ROC_B[m]"]
         AOI_rad = manip.p["AOI[deg]"] * np.pi / 180
+        defocusXA_D = manip.p["defocus[D]"] + manip.p["defocusX[D]"]
+        defocusXB_D = manip.p["defocus_B[D]"] + manip.p["defocusX_B[D]"]
+        defocusYA_D = manip.p["defocus[D]"] + manip.p["defocusY[D]"]
+        defocusYB_D = manip.p["defocus_B[D]"] + manip.p["defocusY_B[D]"]
+        substrate_lens_D = manip.p["substrate_lens[D]"]
+        matrixAD_X = matrix_stack([[1, 0], [defocusXA_D, 1]])
+        matrixAD_Y = matrix_stack([[1, 0], [defocusYA_D, 1]])
+        matrixBD_X = matrix_stack([[1, 0], [defocusXB_D, 1]])
+        matrixBD_Y = matrix_stack([[1, 0], [defocusYB_D, 1]])
 
         # TODO, linear approximation
         depth_m = depth_m / np.cos(AOI_rad * n_A / n_I)
```

Comment:
And shouldn't all of these be something like
```
adj = np.sqrt(1 - (n_A / n_I * np.sin(AOI_rad))**2)
depth_m = depth_m / adj
```
since the propagation depth is L/cos(theta2) where
n_A * sin(AOI_rad) = n_I * sin(theta2)
and cos(theta2) is the above adj.